modules/300-prometheus/templates/prometheus/prometheus.yaml

```diff
@@ -221,7 +218,21 @@ spec:
     {{- end }}
     {{- if .spec.writeRelabelConfigs }}
     writeRelabelConfigs:
-    {{- .spec.writeRelabelConfigs | toYaml | nindent 4 }}
+      {{- .spec.writeRelabelConfigs | toYaml | nindent 6 }}
+      - targetLabel: prometheus
+        replacement: deckhouse
+      {{- with $externalLabels }}
+      {{- range $key, $value := . }}
+      - sourceLabels: [{{ $key }}]
+        targetLabel: __tmp_{{ $key }}_exists
+      - sourceLabels: [__tmp_{{ $key }}_exists]
+        regex: "^$"
+        targetLabel: {{ $key }}
+        replacement: {{ $value | quote }}
+      - target_label: __tmp_{{ $key }}_exists
+        action: drop
+      {{- end }}
+      {{- end }}
```

Comment:
The same question about the mandatory rules is valid here as well.

Also, the 'prometheus' label is forcefully replaced here which contradicts the external labels' semantics.
IMAO, we have to push the `prometheus: deckhouse` pair into the external labels dictionary and generate relabel rules in the range loop.